cmd/export_all_history.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/spf13/cobra"
+
+	"github.com/stellar/stellar-etl/internal/input"
+	"github.com/stellar/stellar-etl/internal/toid"
+	"github.com/stellar/stellar-etl/internal/transform"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+var allHistoryCmd = &cobra.Command{
+	Use:   "export_all_history",
+	Short: "Exports all stellar network history.",
+	Long: `Exports historical stellar network data between provided start-ledger/end-ledger to output files. 
+This is a termporary command used to reduce the amount of requests to history archives 
+in order to mitigate egress costs for the entity hosting history archives.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cmdLogger.SetLevel(logrus.InfoLevel)
+		endNum, strictExport, isTest, isFuture, extra := utils.MustCommonFlags(cmd.Flags(), cmdLogger)
+		cmdLogger.StrictExport = strictExport
+		startNum, path, limit := utils.MustArchiveFlags(cmd.Flags(), cmdLogger)
+		gcsBucket, gcpCredentials := utils.MustGcsFlags(cmd.Flags(), cmdLogger)
+		env := utils.GetEnvironmentDetails(isTest, isFuture)
+
+		allHistory, err := input.GetAllHistory(startNum, endNum, limit, env)
+		if err != nil {
+			cmdLogger.Fatal("could not read all history: ", err)
+		}
+
+		cmdLogger.Info("start doing other exports")
+		getOperations(allHistory.Operations, extra, gcpCredentials, gcsBucket, path+"exported_operations.txt", env)
+		getTrades(allHistory.Trades, extra, gcpCredentials, gcsBucket, path+"exported_trades.txt")
+		getEffects(allHistory.Ledgers, extra, gcpCredentials, gcsBucket, path+"exported_effects.txt", env)
+		getTransactions(allHistory.Ledgers, extra, gcpCredentials, gcsBucket, path+"exported_transactions.txt")
+		getDiagnosticEvents(allHistory.Ledgers, extra, gcpCredentials, gcsBucket, path+"exported_diagnostic_events.txt")
+		cmdLogger.Info("done doing other exports")
+	},
+}
+
+func getOperations(operations []input.OperationTransformInput, extra map[string]string, gcpCredentials string, gcsBucket string, path string, env utils.EnvironmentDetails) {
+	outFileOperations := mustOutFile(path)
+	numFailures := 0
+	totalNumBytes := 0
+	for _, transformInput := range operations {
+		transformed, err := transform.TransformOperation(transformInput.Operation, transformInput.OperationIndex, transformInput.Transaction, transformInput.LedgerSeqNum, transformInput.LedgerCloseMeta, env.NetworkPassphrase)
+		if err != nil {
+			txIndex := transformInput.Transaction.Index
+			cmdLogger.LogError(fmt.Errorf("could not transform operation %d in transaction %d in ledger %d: %v", transformInput.OperationIndex, txIndex, transformInput.LedgerSeqNum, err))
+			numFailures += 1
+			continue
+		}
+
+		numBytes, err := exportEntry(transformed, outFileOperations, extra)
+		if err != nil {
+			cmdLogger.LogError(fmt.Errorf("could not export operation: %v", err))
+			numFailures += 1
+			continue
+		}
+		totalNumBytes += numBytes
+	}
+
+	outFileOperations.Close()
+	cmdLogger.Info("Number of bytes written: ", totalNumBytes)
```

Comment:
If we get frequent transient errors, more logging that identifies which file/export process failed might be helpful. Since this is a temp fix, i'll leave it up to you whether that's worthwhile to add.

Reply:
Yeah it's probably not that useful. Any issue would more easily be resolved by running the original pipeline instead of this hack